Cats and stuff

adelbertc opened the pull request:
It compiles 🤷‍♂️ It has... very minimal tests. It wasn't even using the Scalaz ScalaCheck bindings.

adelbertc committed:
Ermine on the shiny stuff

adelbertc committed:
Formatting and good practices

* sealed abstract class + Product + Serializable
* Proper Scaladoc
* Spacing, indentation, etc
* Remove commented out code

adelbertc committed:
Catsify!

rossabaker commented on build.sbt:
```diff
@@ -2,7 +2,7 @@ name := "parser"
 organization in Global := "io.verizon.ermine"
 
-crossScalaVersions in Global := Seq("2.12.1", "2.11.8", "2.10.6")
+crossScalaVersions in Global := Seq("2.12.4", "2.11.12", "2.10.6")
```
We should bump these in Travis, too.  That's what will be in effect when we publish.

rossabaker commented on build.sbt:
```diff
@@ -20,13 +20,15 @@ scalacOptions ++=
 resolvers += Resolver.bintrayRepo("non", "maven") // for kind-projector
 
-addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.3")
+addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.4")
```
There's an 0.9.5 now.  I don't know why it's better than 0.9.4, but I just bumped it on another project.

rossabaker commented on src/main/scala/scalaparsers/Token.scala:
```diff
@@ -5,21 +5,20 @@ package scalaparsers
+sealed abstract class Token extends Product with Serializable
 
-case object VSemi extends Token {
+final case object VSemi extends Token {
```
I like the final case classes.  Is "final case object" different from "case object" in some subtle way I've not yet discovered?

adelbertc replied:
No clue, I just have a habit/compulsion to tack `final` onto any `case {class,object}`. People seem to like it https://twitter.com/viktorklang/status/735556776989491201 😉

rossabaker replied:
Yeah, I think it's great on case class.  I don't think it matters one way or another on case object.  Mostly asking out of curiosity.

rossabaker reviewed (changes requested):
Mostly LGTM. I'd bump those Scala versions in Travis while we're here.

adelbertc committed:
Bump Travis Scala versions and bump kind-projector to 0.9.5

rossabaker commented:
LGTM.  Leaving the stop sign up while we wait for cats-1.0 and to get rid of the orphan instances with whatever solution it offers.

timperrett commented:
`version.sbt` needs to be updated to the next major version as this release will be entirely incompatible.

adelbertc committed:
Bump version

rossabaker committed:
Crossbuild against Scala 2.10.7

rossabaker committed:
Remove CatsOrphans

Once we get https://github.com/typelevel/cats/pull/2087, we can
revert to the way things were.

rossabaker committed:
cats-1.0.0-RC2
